SRVKS-1243: Add Serving transport encryption docs (#82435)
openshift/openshift-docs: modules/serving-transport-encryption-configuring-selfsigned-clusterissuer.adoc

// Module included in the following assemblies:
//
// * knative-serving/serving-transport-encryption.adoc
:_content-type: PROCEDURE
[id="serving-transport-encryption-configuring-selfsigned-clusterissuer_{context}"]

> = Configuring a SelfSigned cluster issuer

Author: "SelfSigned" and "self-signed" are used slightly differently:
https://cert-manager.io/docs/configuration/selfsigned/

The following procedure uses a `SelfSigned` issuer as the root certificate. See the link:https://cert-manager.io/docs/configuration/selfsigned/[SelfSigned cert-manager documentation] for information about the implications and limitations of this method.

If you manage your own company-specific Private Key Infrastructure (PKI), use the CA issuer. Refer to the link:https://cert-manager.io/docs/configuration/ca/[cert-manager documentation on CA issuers] for more details.

.Procedure

> . Create a `SelfSigned` `ClusterIssuer` custom resource (CR):

Author: Again, "SelfSigned" and "self-signed" are used slightly differently.

> :_content-type: PROCEDURE

Reviewer: Same as previous comments, make sure that proper metadata for modules is included

[id="serving-transport-encryption-configuring-selfsigned-clusterissuer_{context}"]

> = Configuring a SelfSigned ClusterIssuer

Reviewer: ```suggestion
= Configuring a SelfSigned cluster issuer
```
formatting per https://cert-manager.io/docs/configuration/selfsigned/

[IMPORTANT]
====
For simplicity, this procedure uses a `SelfSigned` issuer as root certificate. However, be aware of the implications and limitations of this method documented in the "SelfSigned" section of the `cert-manager` documentation.

If you run your company-specific Private Key Infrastructure (PKI), Red Hat recommends the CA issuer. Refer to the cert-manager documentation on CA issuers for more details. However, you can use any other issuer that allows signing of certificates as described in the "Choice of a certificate issuer" section.

> ====

Reviewer: I don't think that parts like this should be notes - it should be reformatted to be a proper abstract instead. Too many notes like this ends up being distracting and taking away from the distinction that having a note usually provides because they're overused.

> [IMPORTANT]
> ====
> For simplicity, this procedure uses a `SelfSigned` issuer as root certificate. However, be aware of the implications and limitations of this method documented in the "SelfSigned" section of the `cert-manager` documentation.
> 
> If you run your company-specific Private Key Infrastructure (PKI), Red Hat recommends the CA issuer. Refer to the cert-manager documentation on CA issuers for more details. However, you can use any other issuer that allows signing of certificates as described in the "Choice of a certificate issuer" section.
> ====

Reviewer: ```suggestion
The following procedure uses a `SelfSigned` issuer as the root certificate. See the link:https://cert-manager.io/docs/configuration/selfsigned/[SelfSigned cert-manager documentation] for information about the implications and limitations of this method.

If you manage your own company-specific Private Key Infrastructure (PKI), use the CA issuer. Refer to the cert-manager documentation on CA issuers for more details.
====
```
Do not use "RH recommends" per https://redhat-documentation.github.io/supplementary-style-guide/#recommend - instead, direct users to take the recommended action. Also, add links to the documentation here when we mention cert-manager docs - there's no reason not to do this as long as the link isn't an xref inside a module. For links to docs.redhat.com or the cert-manager.io docs, links are permitted.

.Procedure

> . Create a `SelfSigned` `ClusterIssuer`:
> +
> [source,yaml]
> ----
> apiVersion: cert-manager.io/v1
> kind: ClusterIssuer
> metadata:
>   name: knative-serving-selfsigned-issuer
> spec:
>   selfSigned: {}
> ----

Reviewer: ```suggestion
. Create a `SelfSigned` `ClusterIssuer` custom resource (CR):
+
.Example ClusterIssuer CR
[source,yaml]
----
apiVersion: cert-manager.io/v1
kind: ClusterIssuer
metadata:
  name: knative-serving-selfsigned-issuer
spec:
  selfSigned: {}
----
```
- Fix formatting per https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#object-references
- Add code block titles, see examples https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#source-tags-for-terminal-commands-and-output

> . Apply the `ClusterIssuer` resource:
> +
> [source,yaml]
> ----
> $ oc apply -f <filename>
> ----

Reviewer: ```suggestion
. Apply the `ClusterIssuer` CR by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>
----
```

> . Create a root certificate using the previously created `SelfSigned ClusterIssuer`:

Reviewer: ```suggestion
. Create a root certificate that refers to the previously created `ClusterIssuer` CR:
```

> +
> [source,yaml]

Reviewer: ```suggestion
+
.Example root certificate
[source,yaml]
```

----
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: knative-serving-selfsigned-ca
  namespace: cert-manager <1>
spec:
  secretName: knative-serving-ca <2>

  isCA: true
  commonName: selfsigned-ca
  privateKey:
    algorithm: ECDSA
    size: 256

  issuerRef:
    name: knative-serving-selfsigned-issuer
    kind: ClusterIssuer
    group: cert-manager.io
----
<1> The {cert-manager-operator} namespace, `cert-manager` by default.

> <2> Secret name later used for the `ClusterIssuer` for Serving.

Reviewer: ```suggestion
<2> Secret name later used for the `ClusterIssuer` CR for Knative Serving.
```

> . Apply the `Certificate` resource:
> +
> [source,yaml]
> ----
> $ oc apply -f <filename>
> ----

Reviewer: ```suggestion
. Apply the `Certificate` CR by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>
----
```

[id="additional-resources_serving-transport-encryption-configuring-selfsigned-clusterissuer"]
[role="_additional-resources"]
== Additional resources

> * link:https://cert-manager.io/docs/configuration/selfsigned/[OpenShift Serverless Serving Transport Encryption]

Reviewer 2: The title of this link seems to be wrong. It points to Cert Manager docs, not to "OpenShift Serverless Serving Transport Encryption"

> * link:https://cert-manager.io/docs/configuration/ca/[cert-manager documentation on CA issuers]

Reviewer: Rather than including these as additional resources, I would include them as links directly in the text as shown in the comments / example I provided for selfsigned docs. I think having these links directly where they are mentioned since we're able to will be more helpful to users.
